app/webhook.go

```diff
@@ -323,6 +340,21 @@ func (a *App) CreateIncomingWebhookForChannel(creatorId string, channel *model.C
 		return nil, model.NewAppError("CreateIncomingWebhookForChannel", "api.incoming_webhook.invalid_username.app_error", nil, "", http.StatusBadRequest)
 	}
 
+	if len(hook.WhiteIpList) > 0 {
```

Comment:
I'm personally of the mindset that we should tease out the whitelist changes from the hmac changes and submit as separate PRs. This will maximize clarity in the underlying changes.

Reply:
That's fine too. Whitelist is a really tiny functionality which also belongs to webhook authentication. If you think it's ok I could also separate into two final commits at the end one for HMAC signature and the other for whitelist? Please let me know what you opt for.

Reply:
I'd suggest opening a separate PR after this one to track the whitelist changes.

Reply:
Thanks for the response. I will separate the whitelist in a new branch and open a PR.

Reply:
done